1-js/03-code-quality/01-debugging-chrome/article.md

```diff
@@ -1,196 +1,194 @@
+<span class="devtools" style="background-position:-90px -146px"></span> -- bật/tắt tự động tạm dừng trong trường hợp có lỗi.
+: Khi được bật và các công cụ dành cho nhà phát triển đang mở, lỗi tập lệnh sẽ tự động tạm dừng quá trình thực thi. Sau đó, chúng ta có thể phân tích các biến để xem điều gì đã xảy ra. Vì vậy, nếu tập lệnh của chúng ta chết do lỗi, chúng ta có thể mở trình gỡ lỗi, bật tùy chọn này và tải lại trang để xem tập lệnh chết ở đâu và bối cảnh tại thời điểm đó là gì.
 
-```smart header="Continue to here"
-Right click on a line of code opens the context menu with a great option called "Continue to here".
+```smart header="Tiếp tục đến đây"
+Nhấp chuột phải vào một dòng mã sẽ mở menu ngữ cảnh với một tùy chọn tuyệt vời có tên "Tiếp tục đến đây".
 
-That's handy when we want to move multiple steps forward to the line, but we're too lazy to set a breakpoint.
+Điều đó rất hữu ích khi chúng tôi muốn di chuyển nhiều bước về phía trước dòng, nhưng chúng tôi quá lười để đặt điểm dừng.
 ```
 
-## Logging
+## Ghi nhật ký
 
-To output something to console from our code, there's `console.log` function.
+Để xuất nội dung nào đó ra bảng điều khiển từ mã của chúng ta, có chức năng `console.log`.
 
-For instance, this outputs values from `0` to `4` to console:
+Chẳng hạn, điều này xuất các giá trị từ `0` đến `4` sang bảng điều khiển:
 
 ```js run
-// open console to see
+// mở console để thấy
 for (let i = 0; i < 5; i++) {
   console.log("value,", i);
 }
 ```
 
-Regular users don't see that output, it is in the console. To see it, either open the Console panel of developer tools or press `key:Esc` while in another panel: that opens the console at the bottom.
+Người dùng thông thường không nhìn thấy đầu ra đó, nó nằm trong bảng điều khiển. Để xem bảng điều khiển, hãy mở bảng điều khiển của công cụ dành cho nhà phát triển hoặc nhấn `key:Esc` khi đang ở bảng điều khiển khác: mở bảng điều khiển ở dưới cùng.
 
-If we have enough logging in our code, then we can see what's going on from the records, without the debugger.
+Nếu chúng tôi có đủ thông tin đăng nhập vào mã của mình thì chúng tôi có thể xem điều gì đang diễn ra từ các bản ghi mà không cần trình gỡ lỗi.
 
-## Summary
+## Bản tóm tắt
 
-As we can see, there are three main ways to pause a script:
-1. A breakpoint.
-2. The `debugger` statements.
-3. An error (if dev tools are open and the button <span class="devtools" style="background-position:-90px -146px"></span> is "on").
+Như chúng ta có thể thấy, có ba cách chính để tạm dừng tập lệnh:
+1. Một điểm dừng.
+2. Các câu lệnh `debugger`.
+3. Lỗi (nếu công cụ dành cho nhà phát triển đang mở và nút <span class="devtools" style="background-position:-90px -146px"></span> đang "bật").
 
-When paused, we can debug - examine variables and trace the code to see where the execution goes wrong.
+Khi tạm dừng, chúng tôi có thể gỡ lỗi - kiểm tra các biến và theo dõi mã để xem nơi thực thi sai.
```

Comment:
Từ "we" cần dịch là "chúng ta" chứ không phải "chúng tôi".

Reply:
Chỗ này bạn vẫn chưa sửa.

Reply:
"Khi tạm dừng, chúng tôi có thể gỡ lỗi..." <== Chỗ này bạn sửa "chúng ta" thành "chúng tôi" là nhé.